bookkeeper-server/src/test/java/org/apache/bookkeeper/client/MockBookKeeperTestCase.java

```diff
@@ -303,8 +303,12 @@ protected void suspendBookieForceLedgerAcks(BookieId address) {
     }
 
     protected void resumeBookieWriteAcks(BookieId address) {
-        suspendedBookiesForForceLedgerAcks.remove(address);
-        List<Runnable> pendingResponses = deferredBookieForceLedgerResponses.remove(address);
+        List<Runnable> pendingResponses;
+        synchronized (address) {
```

Comment:
I guess it makes sense since it would prevent the possible data race later, in the other location where changes are made. Perhaps adding a code comment would be helpful.

Reply:
Use field `suspendedBookiesForForceLedgerAcks` or just `this` as the object monitor would be ok too, but data race happens on the same `BookieId` instance causes the problem, so I used it as the object monitor. 

I will add a comment. 
Thanks~